Mathlib/RingTheory/Sweedler.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Yunzhou Xie
+-/
+
+import Mathlib.RingTheory.HopfAlgebra
+
+/-!
+# Sweedler notations for co/bi/hopf algebras
+
+-/
+
+namespace SweedlerNotation
+
+open TensorProduct BigOperators Coalgebra HopfAlgebra
+
+section
+
+variable {R A : Type*} [CommSemiring R] [AddCommMonoid A] [Module R A]
+variable {F : Type*} [FunLike F A (A ⊗[R] A)] [LinearMapClass F R A (A ⊗[R] A)] (comul : F)
+
+lemma abstract_comul_exists_repr (x : A) :
```

Comment:
Thanks for this PR! I think this lemma can be generalised to a random element of a tensor product `A ⊗[R] B` instead of `comul` of something, and then this and the next 4 declarations could be moved to `LinearAlgebra.TensorProduct.Basic` and given less `comul`-y names. And then the `Coalgebra` and `HopfAlgebra` sections could just be moved to the `Coalgebra` and `HopfAlgebra` files

Reply:
I have generalized and moved this lemma to `LinearAlgebra.TensorProduct.Basic`, but it seems that the file is too long (